### PR TITLE
Migrate to tox

### PR DESCRIPTION
**Pull request by emlun**
`python setup.py test` is deprecated, see
https://setuptools.readthedocs.io/en/latest/history.html#v41-5-0:

>#1878: Formally deprecated the test command, with the recommendation
that users migrate to tox.

### CODE REVIEW

**Comment by emlun** [comment]
Ah right, the `additional_tests()` hook [used for the integration tests](https://github.com/Yubico/yubikey-manager/blob/88642605dc45b3a1f7e50e053b01f32154777e3b/doc/integration-tests.adoc#how-it-works) is not from unittest but from setuptools, so we can't just replace `python setup.py unittest` with `python -m unittest` without also adding some code to discover and build the test suite. Migrating to pytest would fix that, so let's discuss/take care of that first. We'll probably need to close this again.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# and then run "tox" from this directory.
+
+[tox]
+envlist = py
+
+[testenv]
+# Pure software tests
+# Invoke as `tox`, `tox -e py27` etc.
+deps =
+    -r{toxinidir}/test_requirements.txt
```
Comment (dagheyman) [inline tox.ini]: can the test deps be inlined in the tox file itself?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Tests with real YubiKeys
+# Invoke as `tox -e integration-test`
+basepython = python
+deps =
+    {[testenv]deps}
+    pytest
+commands_pre =
+    {[testenv]commands_pre}
+commands =
+    python setup.py test
```
Comment (dagheyman) [inline tox.ini]: Should this syntax change as well?

Comment (emlun) [reply tox.ini]: Yes, eventually, but right now this remains the only way to run the integration tests.

File: tox.ini (hunk 1 above)
Comment (emlun) [reply tox.ini]: Yes, will do

**Pull request merged**